core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -435,6 +436,20 @@ class LogManager(logDirs: Seq[File],
       cleaner.startup()
   }
 
+  /**
+   * wait all jobs to complete
+   * @param jobs jobs
+   * @return true if all pass. Otherwise, false
+   */
+  private[log] def waitForAllToComplete(jobs: Seq[Future[_]]): Boolean = {
```

Comment:
I would add a callback here for the failures and do the logging in the callback. That way you can keep the name generic.

Reply:
good suggestion. will address it.